controller/version.mli

```diff
@@ -0,0 +1,30 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2018       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(** The type of LSP server versions. *)
+type t
+
+(** The current version of the LSP server. Please update when doing a release. *)
+val current : t
+
+(** Any version *)
+val any : t
+
+(** Convert a version to a string. *)
+val to_string : t -> string
+
+(** Parse a string to a version. *)
+val of_string : string -> t option
+
+(** Parse a string to a version and throw an exception if invalid. *)
+val of_string_exn : string -> t
+
+(** Compare two versions. *)
+val compare : t -> t -> [ `Equal | `Less | `More ]
```

Comment:
Please don't use variants for use cases where they are not required (as discussed in the Dune context)